### PR TITLE
Show matching log

### PR DESCRIPTION
**Pull request by Deren-Liao**
This PR adds the link to the node value. 
By clicking on the link, it filters log to matching items.
screen shot 2017-02-17 at 10 37 05 am

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs [hunk 1]
```diff
@@ -344,6 +349,31 @@ public void FilterLog(string advancedSearchText)
             Reload();
         }
 
+        private void FilterOnTreeNodeValue(ObjectNodeTree node)
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: Can you add a comment for this function?

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs [hunk 2]
```diff
@@ -344,6 +349,31 @@ public void FilterLog(string advancedSearchText)
...
+            if (ShowAdvancedFilter)
+            {
+                newFilter.Insert(0, Environment.NewLine);
+                newFilter.Insert(0, AdvancedFilterText);
+            }
+            else
+            {
+                newFilter.Insert(0, ComposeSimpleFilters());
+            }
+            AdvancedFilterText = newFilter.ToString();
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: I'm a little confused by these two lines.  You are overriding th advanced filter and forcing to show the advnaced filter?

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/TreeViewConverters/JArrayNode.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        /// <summary>
+        /// Initializes a new instance of <seealso cref="JArrayNode"/> class.
+        /// </summary>
+        /// <param name="name">The name to display for this <paramref name="jArrayObj"/>.</param>
+        /// <param name="jArrayObj">A <seealso cref="JArray"/> object to parse.</param>
+        /// <param name="parent">The parent object that owns <paramref name="jArrayObj"/>. </param>
+        public JArrayNode(string name, JArray jArrayObj, ObjectNodeTree parent) 
+            : base(name, jArrayObj, parent) { }
+
+        protected override void ParseObjectTree(object obj)
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/TreeViewConverters/JArrayNode.cs]: Comment?

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/TreeViewConverters/JObjectNode.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        /// <summary>
+        /// Create a new instance of <seealso cref="JObjectNode"/> class.
+        /// </summary>
+        /// <param name="name">The name to display for this object.</param>
+        /// <param name="jObj">A <seealso cref="JObjectNode"/> object to parse.</param>
+        /// <param name="parent">The parent object that owns <paramref name="jObj"/>.</param>
+        public JObjectNode(string name, JObject jObj, ObjectNodeTree parent) 
+            : base(name, jObj, parent) { }
+
+        protected override void ParseObjectTree(object obj)
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/TreeViewConverters/JObjectNode.cs]: comment?

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/TreeViewConverters/ObjectNodeTree.cs [hunk 5]
```diff
@@ -121,36 +138,60 @@ private void ParseObjectTree(object obj)
...
+                }
+                else
+                {
+                    newNode = new ObjectNodeTree(name, obj, this);
+                }
+            }
+
+            if (newNode != null)
+            {
+                Children.Add(newNode);
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/TreeViewConverters/ObjectNodeTree.cs]: you could just move this to line 164?

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs (hunk 1 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs (hunk 2 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: newFilter firstly compose a new filter line. 
Then it appends the new filter line to existing advanced filter text .
In the case this is currently at simple filter view, compose the advanced filter on the simple filters and show the advanced filter.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/TreeViewConverters/ObjectNodeTree.cs (hunk 5 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/TreeViewConverters/ObjectNodeTree.cs]: Good one. I made the change.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/TreeViewConverters/JObjectNode.cs (hunk 4 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/TreeViewConverters/JObjectNode.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/TreeViewConverters/JArrayNode.cs (hunk 3 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/TreeViewConverters/JArrayNode.cs]: Done.

**Comment by Deren-Liao** [comment]
Made changes.  Please take a look again.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs [hunk 6]
```diff
@@ -360,15 +371,20 @@ private void FilterOnTreeNodeValue(ObjectNodeTree node)
                     newFilter.Insert(0, $"{node.FilterLabel}.");
                 }
             }
+
+            // Append the new filter line to existing filter text.
+            // Or to the composed filter if it is currently showing simple filters.
             if (ShowAdvancedFilter)
-            {
-                newFilter.Insert(0, Environment.NewLine);
+            {
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: super nit, added whitespace here and below

Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: That's intended. Adding a line before comment to make  the following code a "segment". It makes clear  the comment is for these lines of code.

**Comment by Deren-Liao** [comment]
Thank you for the quick response.

**Pull request merged**